documentation/src/docs/security/securing-distributed-mode-cluster.md

```diff
@@ -0,0 +1,311 @@
+   $ keytool -keystore controller01.server.jks -alias CARoot -noprompt \
+           -import -file ca-cert -storepass changeit
+
+   # Import the signed server certificate into the keystore.
+   $ keytool -keystore controller01.server.jks -alias controller01 -noprompt \
+           -import -file controller01.pem -storepass changeit
+   ```
+
+5. Export each server's key into a separate .pem file.
+
+   This is a two step process. First, convert the server's keystore in `.jks` format into `.p12` format.
+
+   ```
+   keytool -importkeystore -srckeystore controller01.jks  \
+                           -destkeystore controller01.p12 \
+                           -srcstoretype jks -deststoretype pkcs12 \
+                           -srcstorepass changeit -deststorepass changeit
+   ```
+
+   Then, export the private key of the server into a `.pem` file. Note that the generated .pem file is not protected by a password, even though the key itself is password-protected, as we are using the `-nodes` flag. So, be sure to protect it using operating system's technical controls as well as procedural controls.
+
+   ```
+   openssl pkcs12 -in controller01.p12 -out controller01.key.pem -passin pass:1111_aaaa -nodes
+   ```
+
+Step 5 concludes this stage, and you are now all set to deploy the certificates and other material in Pravega.
+
+The table below lists the key output of this stage. Note that you'll typically need one of each file per Pravega service, but you may share the same file for services collocated on the same host for logistical or cost reasons.
+
+| Files | Example File| Description |
+|:-----:| :---:|:--------|
+| Certificate in PEM format | `controller01.pem` file | The signed certificate to be used by the service.|
+| Private key in PEM format | `controller01.key.pem` file  |  The private key to be used by the service. |
+| Server keystore in JKS format | `controller01.server.jks` file | The keystore file to be used by the service. |
+
+### Deploying certificates and enabling TLS in Pravega
+
+We'll discuss this in the section titled Enabling Security in Pravega, together with other security configuration and setup.
+
+## Enabling Security in Pravega
+
+Here, we discuss about these tasks:
+
+1. Configuring TLS and Auth Parameters on the Server
+2. Configuring TLS and credentials on the client
+3. Having the TLS and Auth parameters take effect
+
+
+### Configuring TLS and Auth Parameters on the Server
+
+This task is about using the certificates, keys, keystores and truststores to configure TLS on the server side, and configuring Auth.
+
+We'll need to configure the following services on the server side:
+
+1. Controller services
+2. Segment Store services
+3. Zookeeper services (Optionally)
+
+Lets' discuss about #1 and #2 below. For information about turning TLS on for Zookeeper, refer to its [SSL](https://cwiki.apache.org/confluence/display/ZOOKEEPER/ZooKeeper+SSL+User+Guide) documentation.
+
+**Controller Service:**
```

Comment:
###

Reply:
Fixed. Changed from "Controller Service" to "Controller".